bibtex-actions.el

```diff
@@ -611,35 +607,25 @@ interactively when `bibtex-actions-at-point-fallback' is non-nil.
 With prefix ARG, rebuild the cache before offering candidates."
   (interactive "P")
   (if (fboundp 'embark-dwim)
-      (let ((embark-keymap-alist '((bibtex . bibtex-actions-map))))
-        (condition-case err
-            (let ((embark-target-finders '(bibtex-actions-citation-key-at-point))
-                  (embark-keymap-alist '((citation-key . bibtex-actions-buffer-map)))
-                  ;; `embark-general-map' is useless in this case
-                  ;; because we are not in the minibuffer and the
-                  ;; TARGET is not string
-                  (embark-general-map embark-meta-map))
-              ;; This is a workaround: unless in the minibuffer,
-              ;; embark cannot pass TARGET to action when the action
-              ;; is an interactive command
-              (cl-letf (((symbol-function 'embark--act)
-                         (lambda (action target _)
-                           (if (where-is-internal action (list bibtex-actions-buffer-map))
-                               (funcall action target)
-                             (command-execute action)))))
-                (if bibtex-actions-embark-dwim
-                    (embark-dwim)
-                  (embark-act))))
-          (user-error
-           (when (and (string-equal (error-message-string err) "No target found")
-                      bibtex-actions-at-point-fallback)
-             (bibtex-actions-run-default-action
-              (bibtex-actions-read :rebuild-cache arg))))))
+      (condition-case err
+          (if bibtex-actions-embark-dwim
+              (embark-dwim)
+            (embark-act))
```

Comment:
Small suggestion here - why not set a `bibtex-actions-at-point-function` to `embark-dwim/embark-act` by default. If this function is not fbound, then call the bibtex-actions-run-default-action?

Reply:
@ilupin - can you answer @minad's question here?

Reply:
@minad - I may be misunderstanding your suggestion here, but my understanding is this is what this function does. 

The variable `bibtex-actions-embark-dwim` is set to `t` by default, which means if one does `M-x bibtex-actions-at-point`, or in org `M-x org-open-at-point` with Embark installed, it will run `embark-dwim`.

Reply:
`bibtex-actions-at-point` was actually added as an entrypoint command equivalent to `ivy/helm-bibtex`, rather than `embark-act/dwim`/`org-open-at-point`, at first. Therefore it is more than an `at-point` function, though the name may be misleading. In this sense, the code can be simplified as suggested.

Reply:
My proposal is to simplify the function as follows:

~~~ elisp
(defvar bibtex-actions-at-point-function 'embark-dwim)
(defvar bibtex-actions-at-point-fallback t)

(defun bibtex-actions-at-point (&optional arg)
  (interactive "P")
  (if (fboundp bibtex-actions-at-point-function)
      (condition-case err
          (funcall bibtex-actions-at-point-function)
        (user-error
         (when (and (string-equal (error-message-string err) "No target found")
                    bibtex-actions-at-point-fallback) ;; fallback when no target is found
           (let ((bibtex-actions-at-point-function)) ;; disable bibtex-actions-at-point-function and retry
             (bibtex-actions-at-point arg)))))
    (bibtex-actions-run-default-action ;; prompt directly when bibtex-actions-at-point-function is nil or not bound
     (bibtex-actions-read :rebuild-cache arg))))
~~~